lib/puppet/reports/datadog_reports.rb

```diff
@@ -110,6 +124,21 @@ def process
       event_data << "\n@@@\n"
     end
 
+    # Check for a running environment other than what is defined for check_environments in parameters
+    if CHECK_ENVIRONMENTS
+      if CHECK_ENVIRONMENTS.include?(@msg_environment) && @noop == false
+        environment_check_status = 0
+        environment_check_message = "#{@msg_host} is configured to use environment production"
```

Comment:
Nit: Is "production" a common puppet term that everyone understands or is this a Krux term?

Reply:
I think the set of people who recognize "production" as a branch name would be a superset of those using r10k. Which is a lot.

But I'm not sure this message is saying what it means. I think it wants to say that host is using *a production environment*, which is defined as any in the list of CHECK_ENVIRONMENTS. (In other words, this could be considered a recovery.) That might mean CHECK_ENVIRONMENTS/`:check_environments` should be called CHECK_PRODUCTION_ENVIRONMENTS, or maybe CHECK_ALLOWED_ENVIRONMENTS.

Reply:
I meant to remove the word production from that message once I made the environments for the check configurable.  Now it's removed and will just say it's configured to use whatever environment it is but that'll be ok status if it's in the list of allowed environments.

Reply:
That being said, `production` is the default environment if one is not configured and it's assumed that it'll always be there.  It's something nearly everyone that uses puppet should be familiar with and is the environment mentioned in the upstream docs that talks about environments, setting up a control repo & r10k, etc